services/iota_wrappers.go

```diff
@@ -458,35 +491,41 @@ func verifyChunksMatchRecord(chunks []models.DataMap, checkTrunkAndBranch bool)
 				continue
 			}
 			if _, ok := transactionObjects[chunkAddress]; ok {
-				matchFound := false
-				for _, txObject := range transactionObjects[chunkAddress] {
-					if chunksMatch(txObject, chunk, checkTrunkAndBranch) {
-						matchFound = true
-						break
-					}
-				}
-				if matchFound {
-					filteredChunks.MatchesTangle = append(filteredChunks.MatchesTangle, chunk)
-				} else {
-					filteredChunks.DoesNotMatchTangle = append(filteredChunks.DoesNotMatchTangle, chunk)
-				}
+
+				matchFound := checkTxObjectsForMatch(transactionObjects[chunkAddress], chunk, checkTrunkAndBranch)
+				assignToChunksArray(matchFound, chunk, &matchesTangle, &doesNotMatch)
 			} else {
-				filteredChunks.NotAttached = append(filteredChunks.NotAttached, chunk)
+				notAttached = append(notAttached, chunk)
 			}
 		}
-	} else if len(response.Hashes) == 0 {
-		filteredChunks.NotAttached = chunks
 	}
+	return matchesTangle, notAttached, doesNotMatch
+}
 
-	if len(filteredChunks.MatchesTangle) > 0 {
-		oyster_utils.LogToSegment("iota_wrappers: chunks_matched_tangle", analytics.NewProperties().
-			Set("num_chunks", len(filteredChunks.MatchesTangle)))
+func assignToChunksArray(matchFound bool, chunk models.DataMap, matchesTangle *[]models.DataMap, doesNotMatch *[]models.DataMap) {
+	if matchFound {
+		*(matchesTangle) = append(*(matchesTangle), chunk)
+	} else {
+		*(doesNotMatch) = append(*(doesNotMatch), chunk)
 	}
-	if len(filteredChunks.NotAttached) > 0 {
-		oyster_utils.LogToSegment("iota_wrappers: not_attached", analytics.NewProperties().
-			Set("num_chunks", len(filteredChunks.NotAttached)))
+}
+
+func checkTxObjectsForMatch(transactionObjectsArray []giota.Transaction, chunk models.DataMap, checkTrunkAndBranch bool) (matchFound bool) {
+	matchFound = false
+	for _, txObject := range transactionObjectsArray {
+		if chunksMatch(txObject, chunk, checkTrunkAndBranch) {
+			matchFound = true
+			break
+		}
 	}
-	return filteredChunks, err
+	return matchFound
+}
+
+func makeTransactionObjects(transactionObjects []giota.Transaction) (transactionObjectsMap map[giota.Address][]giota.Transaction) {
+	for _, txObject := range transactionObjects {
+		transactionObjectsMap[txObject.Address] = append(transactionObjectsMap[txObject.Address], txObject)
```

Comment:
I don't think you create this object, and it will lead to panic. See: https://play.golang.org/p/A9JbB_Q499q

Reply:
Good catch, thanks!